### ISSUE TITLE
[BUG] - install error, deprecated version of the Ads

### ISSUE THREAD

**Issue by AlexandreBroggio** [facebookincubator/Facebook-For-OpenCart#554]
**FBE plugin version**
3.1.2

**OpenCart version**
Versão 3.0.1.1

**Describe the bug**
After installing it displays the error, An error occured while generating system user access token: (#2635) You are calling a deprecated version of the Ads API. Please update to the latest version: v9.0.

Which line can I do this update on?

**Additional context**
I was using another one, she stopped sending the AddCard and Purchase events

**Comment by tudorr89** [comment]
Uninstall the plugin completely, install version 3.0.0 and configure it with your store. then just install latest version 3.1.2 on top of it and it should work. At least it did for me!

**Comment by muhad37** [comment]
> Uninstall the plugin completely, install version 3.0.0 and configure it with your store. then just install latest version 3.1.2 on top of it and it should work. At least it did for me!

Hi, i have same problem 
i tried your idea but its not working 
thanks

**Comment by AlexandreBroggio** [comment]
I tried your idea, but it also didn't work ☹

**Comment by azzisharbel** [comment]
Same problem here, any idea on how to resolve the below issue:
`An error occurred while generating system user access token: (#2635) You are calling a deprecated version of the Ads API. Please update to the latest version: v9.0.`

**Comment by coorgle** [comment]
![Screenshot 2021-03-09 at 10 30 55 AM](https://user-images.githubusercontent.com/44504372/110421707-09b90d80-80c4-11eb-84dd-203902741130.png)
Yeah, same error here

**Comment by Medoubnali** [comment]
Same problem more then 1 week

**Comment by AlexandreBroggio** [comment]
As a temporary thing I configured the event by the pixel using the event configuration tool, placing the purchase event on the thank you page, but it does not take the purchase amount, it only informs you that a purchase has occurred

**Comment by azzisharbel** [comment]
Any suggested ideas at least for a temporary fix ?

**Comment by cafemasasandalye** [comment]
Please fix it. we have same problem

**Comment by coorgle** [comment]
seems like they are least bothered!

**Comment by sebastianvat** [comment]
hii, i have the same problem,
After installing it displays the error, An error occured while generating system user access token: (#2635) You are calling a deprecated version of the Ads API. Please update to the latest version: v9.0.


and an older version is not instaling...

Do you have anithing?


i also have this in the log document


2021-03-14 17:51:10 - PHP version = 7.3.11
2021-03-14 17:51:10 - Verifying Availability of Facebook Ads Extension Settings Availability
2021-03-14 17:51:10 - Pixel ID = false
2021-03-14 17:51:10 - Pixel use PII = false
2021-03-14 17:51:10 - Use S2S = false
2021-03-14 17:51:10 - Page ID = false
2021-03-14 17:51:10 - System User Access Token = false
2021-03-14 17:51:10 - Messenger plugin = false
2021-03-14 17:51:10 - Messenger customization locale = false

**Comment by microtoy** [comment]
> ![Screenshot 2021-03-09 at 10 30 55 AM](https://user-images.githubusercontent.com/44504372/110421707-09b90d80-80c4-11eb-84dd-203902741130.png)
> Yeah, same error here

Exactly the same issue!!!!

**Comment by serkanbalta** [comment]
Same problem goes on all over the Opencart 3x

**Comment by haxcop** [comment]
Well, I guess I arrived too late for this party... Facebook could you please allow us to know how to fix this issue?

**Comment by coorgle** [comment]
I think this has to be reported to Opencart only because the API or Facebook is hosted by Opencart (facebook.opencart.com) so opencart has to update their APIs i think. will open a forum question in OC forum, everyone please reply there too!

**Comment by coorgle** [comment]
https://forum.opencart.com/viewtopic.php?f=198&t=223017&p=817081&hilit=facebook#p817081

**Comment by Khnaz35** [comment]
They are not replying here, neither on Opencart marketplace looks like the developer is sick or facebook didn't clear his/her payments 🥇

**Comment by Khnaz35** [comment]
Some of the info may be found here: 
https://developers.facebook.com/blog/post/2020/11/10/introducing-graph-v9-marketing-api-v9/

if someone really care can update from here: 
https://developers.facebook.com/docs/graph-api/changelog/version9.0

**Comment by coorgle** [comment]
We have escalated this matter to Facebook's extension development team directly, Will update once we get any response from them.

**Comment by dtying** [comment]
Hi, this issue was due to the API version used was deprecated. As there'll be a new version which is scheduled to be released by end of the March, this issue should be covered and fixed in that release.

**Comment by Khnaz35** [comment]
> We have escalated this matter to Facebook's extension development team directly, Will update once we get any response from them.

From which channel?
Even i have open ticket via opencart platform and waiting for replies from team.

**Comment by mihainita806** [comment]
has someone pass this access token problem?

**Comment by coorgle** [comment]
We think the update will be further delayed, We have a different module for Pixel integration, Let us know if anyone wants to use this module, We will charge only 5USD. For getting this raise a ticket at https://cws.coorgle.com/submitticket.php?step=2&deptid=10

**Comment by cartbinder** [comment]
> Hi, this issue was due to the API version used was deprecated. As there'll be a new version that is scheduled to be released by end of the March, this issue should be covered and fixed in that release.

When the new update will be released?

**Comment by shanebemedia** [comment]
> We think the update will be further delayed, We have a different module for Pixel integration, Let us know if anyone wants to use this module, We will charge only 5USD. For getting this raise a ticket at https://cws.coorgle.com/submitticket.php?step=2&deptid=10

Coorgle will that allow dynamic product feed if installed?

**Comment by coorgle** [comment]
> > We think the update will be further delayed, We have a different module for Pixel integration, Let us know if anyone wants to use this module, We will charge only 5USD. For getting this raise a ticket at https://cws.coorgle.com/submitticket.php?step=2&deptid=10
> 
> Coorgle will that allow dynamic product feed if installed?

Yes, Please register yourself on https://cws.coorgle.com/register.php and raise a ticket with the version, we are giving it for free, as we understand the covid situation and how it has been affecting everyone's sales. You can donate us if you can to coorgle.com@gmail.com 

www.Cartin.store

**Comment by coorgle** [comment]
For Free Update and fix contact us, we are giving Pixel integration for free! register here https://cws.coorgle.com/register.php and raise a ticket for the module with your opencart version

**Comment by yao-tx** [comment]
Hi guys,

Please download the latest version (4.0.1) of the Facebook Business Extension plugin for OpenCart at [OpenCart Marketplace](https://www.opencart.com/index.php?route=marketplace/extension/info&extension_id=32336). This should resolve the issues you are having with the current versions.

If you still have any further issues, please feel free to reach out to us via our help desk [here](http://marketinsg.zendesk.com/) so that we can assist you with your issues.

**Comment by Fydon** [comment]
Thank you @yao-tx. Is the latest source code going to be pushed here?

**Comment by yao-tx** [comment]
> Thank you @yao-tx. Is the latest source code going to be pushed here?

Yes, we will be pushing the source code onto GitHub soon.

**Comment by mihainita806** [comment]
Hello, i am writing to you because I noticed 2 problems with the pixel.

The first one is in the purchase event. This event is missing some parameters. My site works like this: after checkout when you press the confirm button the "thank you for your order page" loads. I've attached a photo for you to exactly see what is all about.

The second one is similar but is less important. The add to cart content_ids and contents parameters have the ID and the quantity like this ["Product ID", "Quantity"] but the Quantity parameter has some conflict with the product id.

The checkout event works just fine.

I am using the last version of the OpenCart Facebook Plugin v4.0.1
Thank you and I appreciate a quick fix for the first problem.
Mihai Nita
